src/logic/common/util/CommonUtils.js

```diff
@@ -49,6 +55,37 @@ function hideContent($content, contentText, spoilersArr, siteConfig) {
 	}
 }
 
+function evaluateTokenArr(text, arr) {
+	console.log("TOKEN ARR: " + JSON.stringify(arr));
+ 
+	let stack = [];
+	let bool1, bool2;
+
+	for (let i=0; i<arr.length; i++) {
+		
+		if (arr[i]["tokenType"] === "LITERAL") {
+			stack.push(text.indexOf(arr[i]["value"]) !== -1);
+		}
+		else {
+			bool1 = stack.pop();
+			bool2 = stack.pop();
+
+			if (arr[i]["value"] === "AND") {
+				stack.push(bool1 && bool2);
+			}
+			else if (arr[i]["value"] === "OR") {
+				stack.push(bool1 || bool2);
+			}
+			else {
+				throw new Error("Internal error: BINARY_OP does not have value AND or OR");
+			}
+		}
+
+	}
+
+	return stack.pop();
```

Comment:
How would we display the error? Panel is most likely closed while this code is running.